Do not declare `wp-prettier` as a dependency of `@wordpress/eslint-plugin`

## What problem does this address?
<!--
Please describe if this feature or enhancement is related to a current problem
or pain point. For example, "I'm always frustrated when ..." or "It is currently
difficult to ...".
-->

Currently, the `@wordpress/eslint-plugin` is declaring the WordPress fork of prettier as direct dependency and therefore making the wrong assumption that every project using that package is going to use WP fork of prettier.

At 10up, we have our own [build tool](https://github.com/10up/10up-toolkit) similar to `wp-scripts` but we do not use `wp-prettier`, we use prettier original implementation. Our eslint config exports a `wordpress` config that makes use of the custom `@wordpress` eslint rules but depending on `@wordpress/eslint-plugin` causes some issues (from npm ERESOLVE warnings to errors in some versions of npm) due to conflicting dependencies of prettier.

IMO prettier and wp-prettier have nothing to do with `@wordpress/eslint-plugin` and it should be up to the consumers of the plugin to decide how they want to format their code.

## What is your proposed solution?
<!--
Please outline the feature or enhancement that you want and how it addresses any
problem identified above.
-->

Remove wp-prettier from `package.json` in the eslint-plugin package https://github.com/WordPress/gutenberg/blob/ab65bf431643e3cfc237b6634b68a01bae0b81f4/packages/eslint-plugin/package.json#L49 and have `wp-scripts` declare prettier's fork as a direct dependency.

as FYI this is what npm ls look like for prettier when using npm 8 on 10up-toolkit. I've had reports where npm install completely fails when using npm 7. npm 8 just gives an ERESOLVE warning.

<img width="971" alt="image" src="https://user-images.githubusercontent.com/6104632/156882400-5106121a-3381-4258-9acb-4465278e3c63.png">

If consumers of `@wordpress/eslint-plugin` are already having issues with the alias and are having to add the alias manually I'm guessing we could just remove the alias from `@wordpress/eslint-plugin`?

--- a/packages/eslint-plugin/CHANGELOG.md
+++ b/packages/eslint-plugin/CHANGELOG.md
@@ -2,7 +2,13 @@
 
 ## Unreleased
 
-–   Replaced no-shadow eslint rule with @typescript-eslint/no-shadow ([#38665](https://github.com/WordPress/gutenberg/pull/38665)).
+### Breaking Changes
+
+-   The integration with [Prettier](https://prettier.io) is now optional and gets activated when the `prettier` package is installed in the project ([#39244](https://github.com/WordPress/gutenberg/pull/39244)).
+
+### Bug Fix
+
+-   Replaced no-shadow eslint rule with @typescript-eslint/no-shadow ([#38665](https://github.com/WordPress/gutenberg/pull/38665)).
 
 ## 10.0.0 (2022-01-27)
 

--- a/packages/eslint-plugin/README.md
+++ b/packages/eslint-plugin/README.md
@@ -24,9 +24,21 @@ To opt-in to the default configuration, extend your own project's `.eslintrc` fi
 
 Refer to the [ESLint documentation on Shareable Configs](http://eslint.org/docs/developer-guide/shareable-configs) for more information.
 
-The `recommended` preset will include rules governing an ES2015+ environment, and includes rules from the [`eslint-plugin-jsx-a11y`](https://github.com/evcohen/eslint-plugin-jsx-a11y), [`eslint-plugin-react`](https://github.com/yannickcr/eslint-plugin-react), and [`eslint-plugin-prettier`](https://github.com/prettier/eslint-plugin-prettier) projects. It also includes an optional integration with [`@typescript-eslint/eslint-plugin`](https://github.com/typescript-eslint/typescript-eslint) that gets activated when the [`typescript`](https://www.npmjs.com/package/typescript) package is installed in the project.
+The `recommended` preset will include rules governing an ES2015+ environment, and includes rules from the [`eslint-plugin-jsdoc`](https://github.com/gajus/eslint-plugin-jsdoc), [`eslint-plugin-jsx-a11y`](https://github.com/evcohen/eslint-plugin-jsx-a11y), [`eslint-plugin-react`](https://github.com/yannickcr/eslint-plugin-react), and other similar plugins.
 
-There is also `recommended-with-formatting` ruleset for projects that want to opt out from [Prettier](https://prettier.io). It has the native ESLint code formatting rules enabled instead.
+This preset offers an optional integration with the [`eslint-plugin-prettier`](https://github.com/prettier/eslint-plugin-prettier) package that runs [Prettier](https://prettier.io) code formatter and reports differences as individual ESLint issues. You can activate it by installing the [`prettier`](https://www.npmjs.com/package/prettier) package separately with:
+
+```bash
+npm install prettier --save-dev
+```
+
+Finally, this ruleset also includes an optional integration with the [`@typescript-eslint/eslint-plugin`](https://github.com/typescript-eslint/typescript-eslint) package that enables ESLint to support [TypeScript](https://www.typescriptlang.org) language. You can activate it by installing the [`typescript`](https://www.npmjs.com/package/typescript) package separately with:
+
+```bash
+npm install typescript --save-dev
+```
+
+There is also `recommended-with-formatting` ruleset for projects that want to ensure that [Prettier](https://prettier.io) and [TypeScript](https://www.typescriptlang.org) integration is never activated. This preset has the native ESLint code formatting rules enabled instead.
 
 ### Rulesets
 

--- a/packages/eslint-plugin/configs/recommended.js
+++ b/packages/eslint-plugin/configs/recommended.js
@@ -13,19 +13,20 @@ const defaultPrettierConfig = require( '@wordpress/prettier-config' );
  */
 const { isPackageInstalled } = require( '../utils' );
 
-const { config: localPrettierConfig } =
-	cosmiconfigSync( 'prettier' ).search() || {};
-const prettierConfig = { ...defaultPrettierConfig, ...localPrettierConfig };
-
 const config = {
-	extends: [
-		require.resolve( './recommended-with-formatting.js' ),
-		'plugin:prettier/recommended',
-	],
-	rules: {
-		'prettier/prettier': [ 'error', prettierConfig ],
-	},
+	extends: [ require.resolve( './recommended-with-formatting.js' ) ],
 };
+
+if ( isPackageInstalled( 'prettier' ) ) {
+	config.extends.push( 'plugin:prettier/recommended' );
+
+	const { config: localPrettierConfig } =
+		cosmiconfigSync( 'prettier' ).search() || {};
+	const prettierConfig = { ...defaultPrettierConfig, ...localPrettierConfig };
+	config.rules = {
+		'prettier/prettier': [ 'error', prettierConfig ],
+	};
+}
 
 if ( isPackageInstalled( 'typescript' ) ) {
 	config.settings = {

--- a/packages/eslint-plugin/package.json
+++ b/packages/eslint-plugin/package.json
@@ -46,15 +46,18 @@
 		"eslint-plugin-react": "^7.27.0",
 		"eslint-plugin-react-hooks": "^4.3.0",
 		"globals": "^13.12.0",
-		"prettier": "npm:wp-prettier@2.2.1-beta-1",
 		"requireindex": "^1.2.0"
 	},
 	"peerDependencies": {
 		"@babel/core": ">=7",
 		"eslint": ">=8",
+		"prettier": ">=2",
 		"typescript": ">=4"
 	},
 	"peerDependenciesMeta": {
+		"prettier": {
+			"optional": true
+		},
 		"typescript": {
 			"optional": true
 		}

